Assets/Mirror/Runtime/NetworkConnectionToClient.cs

```diff
@@ -6,16 +6,19 @@ namespace Mirror
 {
     public class NetworkConnectionToClient : NetworkConnection
     {
-        public NetworkConnectionToClient(int networkConnectionId) : base(networkConnectionId)
-        {
-        }
+        public NetworkConnectionToClient(int networkConnectionId) : base(networkConnectionId) { }
 
         public override string address => Transport.activeTransport.ServerGetClientAddress(connectionId);
 
         // internal because no one except Mirror should send bytes directly to
         // the client. they would be detected as a message. send messages instead.
         readonly List<int> singleConnectionId = new List<int> { -1 };
 
+        // Failsafe to kick clients that have stopped sending anything to the server.
+        // Clients ping the server every 2 seconds but transports are unreliable
+        // when it comes to properly generating Disconnect messages to the server.
+        internal override bool IsClientAlive() => Time.time - lastMessageTime < serverIdleTimeout;
```

Comment:
this is kinda odd to think about when it comes to time.
what about Time.time >= lastMessageTime + Timeout?
not sure about you but this seems a lot easier for my brain

Reply:
Then why isn't NetworkTransform written that way.  I think it's an overflow risk to do addition instead of subtraction.

Reply:
resolved to leave it as is in Discord

Reply:
resolved in Discord